examples/example.cpp

```diff
@@ -255,14 +201,19 @@ Node* newObject(std::string typ) {
 }
 
 int main() {
```

Comment:
Unfortunately, those templates break `auto` magic and the example becomes needlessly hard to read.

Reply:
What do you mean by breaking `auto` magic?

Reply:
It requires to explicitly write all types since they will be a part of the template type.

Reply:
Currently it looks like this:
```cpp
auto ctx = impl->NewContext();
```
with `unique_ptr` I will need to write the full type on the right.

Reply:
Ah, yes, of course, but, while it could be annoying to do it generally (you could use `decltype` on the other hand to auto infer the return type and wrap it into an `unique_ptr`) in the case of a file whose purpose is to be an example being explicit about the types is not a bad thing.